### PR TITLE
prepare for VS2019: detect VS installation via COM

### PR DESCRIPTION
**Pull request by rainers**
VS2019 Preview doesn't have a simple registry entry as VS2017 anymore. The COM-API exists for VS2017, too.

Also removed unused VisualStudioVersion.

### CODE REVIEW

File: src/dmd/link.d [hunk 1]
```diff
@@ -1484,4 +1475,87 @@ version (Windows)
...
+        int GetInstallationPath(BSTR* pbstrInstallationPath);
+        int GetInstallationVersion(BSTR* pbstrInstallationVersion);
+        int GetDisplayName(LCID lcid, BSTR* pbstrDisplayName);
+        int GetDescription(LCID lcid, BSTR* pbstrDescription);
+        int ResolvePath(LPCOLESTR pwszRelativePath, BSTR* pbstrAbsolutePath);
+    }
+
+    interface IEnumSetupInstances : IUnknown
+    {
+        // static const GUID iid = uuid("6380BCFF-41D3-4B2E-8B2E-BF8A6810C848");
```
Comment (jacob-carlborg) [inline src/dmd/link.d]: This interface doesn't need an `iid`?

Comment (rainers) [reply src/dmd/link.d]: No, it's returned directly by `ISetupConfiguration.EnumInstances`. `ISetupInstance.iid` is also not needed, but I had already converted it before I noticed.

**Review by wilzbach — APPROVED** [review]
LGTM from looking over these changes. Of course, it would be awesome to have an actual test for it.
I see that it should be available on AppVeyor as [preview](https://www.appveyor.com/updates/2018/12/13/) though it's probably not worth the hassle until it has been released.

**Comment by rainers** [comment]
Yes, it's probably not yet worth setting up a CI to just test this PR or even keep it running just to test the preview.

Maybe I could remove the registry entry that is so far used to detect VS2017 and add a test that removes all environment variables that might disable detection. That could be troublesome to build locally, though.

**Pull request merged**